File: 1716_calculate_money_in_leetcode_bank/solution.php
<?php
// LeetCode 1716 - Calculate Money in Leetcode Bank
// https://leetcode.com/problems/calculate-money-in-leetcode-bank/

class Solution {
    /**
     * @param Integer $n
     * @return Integer
     */
    function totalMoney($n) {
        $weeks = intdiv($n, 7);
        $days = $n % 7;
        return $weeks * 28 + intdiv(7 * $weeks * ($weeks - 1), 2) + $days * ($weeks + 1) + intdiv($days * ($days - 1), 2);
    }
}

File: 1716_calculate_money_in_leetcode_bank/Solution.java
// LeetCode 1716 - Calculate Money in Leetcode Bank
// https://leetcode.com/problems/calculate-money-in-leetcode-bank/

class Solution {
    public int totalMoney(int n) {
        int weeks = n / 7;
        int days = n % 7;
        return weeks * 28 + 7 * weeks * (weeks - 1) / 2 + days * (weeks + 1) + days * (days - 1) / 2;
    }
}
